### PR TITLE
Feature Reservation Model and Controller

### LINKED ISSUE #6: [4pt] Reservation Model and Controller
- [x]  Create Reservation Model
- [x]  Create a Reservation Controller
    - [x] Index Method
    - [x] Create Method

### LINKED ISSUE #7: [4pt] Reservation Routes and Endpoints
- [x] Add Reservation Routes
- [x] Setup /POST endpoint
- [x] Setup /GET endpoint

### PR DESCRIPTION
**Pull request by 78fatemanazari**
### Description
- Create reservation model
- Create reservation controller
- Create reservation endpoints
- Include jwt helper function 
- Create associations between user and reservation model

### CODE REVIEW

File: app/models/reservation.rb
Comment (demesameneshoa) [inline app/models/reservation.rb:1]: - [ ] Please add a validation to the model attributes, so that the api can catch any wrongful inputs

**Review by demesameneshoa — CHANGES_REQUESTED** [review]
Hi @78fatemanazari 🙋‍♂️, 

# ♻️    Status: Changes Required ⏺️  

You have done a great job on this ticket 🎫 , keep it up 👍 but there are some changes you need to make, you are almost there

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 
- The User Authentication is implemented properly ✅ 


## Optional suggestionsgp

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: app/models/reservation.rb
Comment (78fatemanazari) [reply app/models/reservation.rb:1]: Thank you so much @demesameneshoa 
I will make these changes!

**Comment by demesameneshoa** [comment]
Hi @78fatemanazari , 

Your ticket is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
-  Model, Controller and End points created properly  ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------

**Review by demesameneshoa — APPROVED** [review]
Hi @78fatemanazari , 

Your ticket is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
-  Model, Controller and End points created properly  ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------

**Pull request merged**